doc/lspconfig.txt

```diff
@@ -1,242 +1,659 @@
 - rename
 - format
 - refactor
-Neovim provides an interface for all of these features, and the language server
-client is designed to be highly extensible to allow plugins to integrate
-language server features which are not yet present in Neovim core such as [**auto**-completion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion)
-(as opposed to manual completion with omnifunc) and [snippet integration](https://github.com/neovim/nvim-lspconfig/wiki/Snippets).
-These features are not implemented in this repo, but in Neovim core. See `:help
-lsp` for more details.
+
+These features are implemented in neovim core, not lspconfig. See `:help lsp`
+for more details.
+
+NOTE: Feature availability depends on the implementation details of the
+server. that servers may implement only a subset of these features. Always
+consult the server documentation before filing a bug report on a missing
+feature.
+
 ==============================================================================
-3. Install                                                 *lspconfig-install*
+QUICKSTART                                                *lspconfig-quickstart*
 
+- ensure the server is installed and executable from the command line
 
-* Requires [Neovim v0.5.0](https://github.com/neovim/neovim/releases/tag/v0.5.0) or [Nightly](https://github.com/neovim/neovim/releases/tag/nightly). Update Neovim and nvim-lspconfig before reporting an issue.
-* Install nvim-lspconfig like any other Vim plugin, e.g. with [vim-plug](https://github.com/junegunn/vim-plug):
+- enable the server in your neovim configuration (lua example):
+>
+    require'lspconfig'.clangd.setup{}
+<
+- create a new project, ensure it contains a root marker which matches the
+  server requirements specified in `servers.md`.
+
+- open a file within that project, such as `main.c`.
 
-    ```vim
-    Plug 'neovim/nvim-lspconfig'
-    ```
 ==============================================================================
-4. Quickstart                                           *lspconfig-quickstart*
+SETUP                                                          *lspconfig-setup*
+
+The main entrypoint to enabling and customizing a language server is the
+`setup {}` call. Using the defaults for a server built-into lspconfig is
+simple:
+>
+    require'lspconfig'.clangd.setup{}
+<
+The available server names are listed in `./server.md` and match the server
+name in `config.SERVER_NAME` defined in each configuration's source file.
+
+The purpose of `setup{}` is to wrap the call to `vim.lsp.start_client`  which
+is built into neovim core in an autocommand that calls `start_client` or
+`vim.lsp.buf_attach_client()` depending on if the current file appears to be
+in a project that contains a server previously started by lspconfig. See
+|root_dir| for more details.
+
+The `setup{}` function takes a table which contains a superset of the keys
+listed in `:help vim.lsp.start_client()` with the following unique entries:
```

Comment:
I don't think this obvious when you have actually split some options into the override section and.